source/azure_iot_provisioning_client.c

```diff
@@ -0,0 +1,963 @@
+/* Copyright (c) Microsoft Corporation. All rights reserved. */
+/* SPDX-License-Identifier: MIT */
+
+/**
+ * @file azure_iot_provisioning_client.c
+ * @brief Implementation of Azure IoT Provisioning client.
+ * 
+ */
+
+#include "azure_iot_provisioning_client.h"
+
+/* Kernel includes. */
+#include "FreeRTOS.h"
+#include "task.h"
+
+/* azure iot includes. */
+#include "azure_iot_mqtt.h"
+#include "azure/az_iot.h"
+#include "azure/core/az_json.h"
+#include "azure/core/az_version.h"
+
+#ifndef azureiotprovisioningDEFAULT_TOKEN_TIMEOUT_IN_SEC
+    #define azureiotprovisioningDEFAULT_TOKEN_TIMEOUT_IN_SEC    azureiotDEFAULT_TOKEN_TIMEOUT_IN_SEC
+#endif /* azureiotprovisioningDEFAULT_TOKEN_TIMEOUT_IN_SEC */
+
+#ifndef azureiotprovisioningKEEP_ALIVE_TIMEOUT_SECONDS
+    #define azureiotprovisioningKEEP_ALIVE_TIMEOUT_SECONDS    azureiotKEEP_ALIVE_TIMEOUT_SECONDS
+#endif /* azureiotprovisioningKEEP_ALIVE_TIMEOUT_SECONDS */
+
+#ifndef azureiotprovisioningCONNACK_RECV_TIMEOUT_MS
+    #define azureiotprovisioningCONNACK_RECV_TIMEOUT_MS    azureiotCONNACK_RECV_TIMEOUT_MS
+#endif /* azureiotprovisioningCONNACK_RECV_TIMEOUT_MS */
+
+#ifndef azureiotprovisioningUSER_AGENT
+    #define azureiotprovisioningUSER_AGENT             ""
+#endif /* azureiotprovisioningUSER_AGENT */
+
+#define azureiotprovisioningPROCESS_LOOP_TIMEOUT_MS    ( 500U )
+
+#define azureiotprovisioningWF_STATE_INIT              ( 0x0 )
+#define azureiotprovisioningWF_STATE_CONNECT           ( 0x1 )
+#define azureiotprovisioningWF_STATE_SUBSCRIBE         ( 0x2 )
+#define azureiotprovisioningWF_STATE_REQUEST           ( 0x3 )
+#define azureiotprovisioningWF_STATE_RESPONSE          ( 0x4 )
+#define azureiotprovisioningWF_STATE_SUBSCRIBING       ( 0x5 )
+#define azureiotprovisioningWF_STATE_REQUESTING        ( 0x6 )
+#define azureiotprovisioningWF_STATE_WAITING           ( 0x7 )
+#define azureiotprovisioningWF_STATE_COMPLETE          ( 0x8 )
+
+#define azureMAX( a, b )    ( ( a ) > ( b ) ? ( a ) : ( b ) )
+
+#define azureiotprovisioningREQUEST_PAYLOAD_LABEL         "payload"
+#define azureiotprovisioningREQUEST_REGISTRATION_LABEL    "registrationId"
+
+/*-----------------------------------------------------------*/
+
+/**
+ *
+ * State transitions :
+ *          CONNECT -> { SUBSCRIBE | COMPLETE } -> { SUBSCRIBING | COMPLETE } -> { REQUEST | COMPLETE } ->
+ *          REQUESTING -> { RESPONSE | COMPLETE } -> { WAITING | COMPLETE } -> { REQUEST | COMPLETE }
+ *
+ * Note : At COMPLETE state we should check the status of lastOperation to see if WorkFlow( WF )
+ *        finshed without error.
+ *
+ **/
+static void prvAzureIoTProvisioningClientUpdateState( AzureIoTProvisioningClientHandle_t xAzureIoTProvisioningClientHandle,
```

Comment:
Nit/consider: Having to prefix static functions in this file with prvAzureIoTProvisioningClient* is making them pretty long given this isn't public facing.  Any chance we could (well we obviously could:)) / should change these to be shorter?  

The only place that could get confused would be I think on source editors doing function lookup, but even getting down to prvProvClient* would make that unlikely.

Reply:
I renamed all with prefix prvProvClient

---
In reply to: [602629054](https://github.com/Azure/azure-iot-middleware-freertos/pull/49#discussion_r602629054) [](ancestors = 602629054)